test/Generator.Tests/MapDateOnlyConverter.UnitTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Generator.Tests;
+
+[TestClass]
+public class MapDateOnlyConverterUnitTests
+{
+    private readonly JsonSerializerOptions options = new JsonSerializerOptions { DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull };
+
+    [TestMethod]
+    public void PopulatedMapDateOnlyPropertyDeserializesCorrectly()
+    {
+        var json = $"{{\"$dtId\":\"d8985302-4ee1-4a10-b2f5-e854e1682422\",\"assetTag\":\"12345\",\"name\":\"Test Asset\",\"serialNumber\":\"SN12345\",\"runtimeDetails\":{{\"Friday\":\"2020-03-10\",\"Sunday\":\"2020-03-12\"}},\"$metadata\":{{\"$model\":\"{Asset.ModelId}\"}}}}";
+        var expectedAsset = new Asset
+        {
+            Id = "d8985302-4ee1-4a10-b2f5-e854e1682422",
+            AssetTag = "ABC123",
+            Name = "Test Asset",
+            SerialNumber = "SN12345",
+            RuntimeDetails = new Dictionary<string, DateOnly>()
+            {
+                ["Friday"] = new DateOnly(2020, 03, 10),
```

Comment:
This is a rather uncommon way to instantiate a key/value pair and isn't consistent with other ways we do it. 

try:
``` csharp
{
    { "Friday", new DateOnly(2020, 03, 10) },
    { "Sunday", new DateOnly(2020, 03, 12) }
}
```